doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -314,6 +343,36 @@ messages and additional metadata helpful for debugging.
 | `error`        | [`text`]                        | If the source is in an error state, the error message.                                                             |
 | `details`      | [`jsonb`]                       | Additional metadata provided by the source. In case of error, may contain a `hint` field with helpful suggestions. |
 
+<!-- ### `mz_statement_execution_history` -->
+
+<!-- The `mz_statement_execution_history` table contains a row for each -->
+<!-- statement executed, that the system decided to log. Entries older than -->
+<!-- thirty days may be removed. -->
+
+<!-- The system chooses to log statements randomly; the probability of -->
+<!-- logging an execution is controlled by the -->
+<!-- `statement_logging_sample_rate` session variable. A value of 0 means -->
+<!-- to log nothing; a value of 0.8 means to log approximately 80% of -->
+<!-- statement executions. If `statement_logging_sample_rate` is higher -->
+<!-- than `statement_logging_max_sample_rate` (which is set by Materialize -->
+<!-- and cannot be changed by users), the latter is used instead. -->
+
+<!-- | Field                   | Type                         | Meaning                                                                                                                                                                                                                        | -->
+<!-- |-------------------------|------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------| -->
+<!-- | `id`                    | [`uuid`]                     | The ID of the execution event.                                                                                                                                                                                                 | -->
+<!-- | `prepared_statement_id` | [`uuid`]                     | The ID of the prepared statement being executed. Corresponds to [`mz_prepared_statement_history.id`](#mz_prepared_statement_history).                                                                                          | -->
+<!-- | `sample_rate`           | [`double precision`]         | The sampling rate at the time the execution began.                                                                                                                                                                             | -->
```

Comment:
So this is the value of the `SystemVar` at the specific point in time?

Tying this back to the discussion above, could a similar approach be a possible solution to the `application_name` session var (I understand that this has implications on the memory requirements)?